pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -3862,49 +3876,50 @@ public void revertReplaceSegments(String tableNameWithType, String segmentLineag
    * @param lineageUpdateType
    * @param customMap
    */
-  private boolean writeLineageEntryWithTightLoop(TableConfig tableConfig, String lineageEntryId,
+  private boolean writeLineageEntryWithLock(TableConfig tableConfig, String lineageEntryId,
       LineageEntry lineageEntryToUpdate, LineageEntry lineageEntryToMatch, ZkHelixPropertyStore<ZNRecord> propertyStore,
       LineageUpdateType lineageUpdateType, Map<String, String> customMap) {
-    for (int i = 0; i < DEFAULT_SEGMENT_LINEAGE_UPDATE_NUM_RETRY; i++) {
-      // Fetch the segment lineage
-      ZNRecord segmentLineageToUpdateZNRecord =
-          SegmentLineageAccessHelper.getSegmentLineageZNRecord(propertyStore, tableConfig.getTableName());
-      int expectedVersion = segmentLineageToUpdateZNRecord.getVersion();
-      SegmentLineage segmentLineageToUpdate = SegmentLineage.fromZNRecord(segmentLineageToUpdateZNRecord);
-      LineageEntry currentLineageEntry = segmentLineageToUpdate.getLineageEntry(lineageEntryId);
-
-      // If the lineage entry doesn't match with the previously fetched lineage, we need to fail the request.
-      if (!currentLineageEntry.equals(lineageEntryToMatch)) {
-        String errorMsg = String.format(
-            "Aborting the to update lineage entry since we find that the entry has been modified for table %s, "
-                + "entry id: %s", tableConfig.getTableName(), lineageEntryId);
-        LOGGER.error(errorMsg);
-        throw new RuntimeException(errorMsg);
-      }
+    String tableNameWithType = tableConfig.getTableName();
```

Comment:
Considering the distributed update from other controllers, I feel it's still better to put a smaller number of retries here?